certification/pyxis/submit_test.go

```diff
@@ -14,54 +14,83 @@ var _ = Describe("Pyxis Submit", func() {
+		certInput = certificationInput{
+			CertProject: &CertProject{CertificationStatus: "Started"},
+			CertImage: &CertImage{
+				Repositories: []Repository{
+					{
+						Registry:   "my.registry",
+						Repository: "my/repo",
+					},
+				},
+				DockerImageDigest: "sha256:deadb33f",
+			},
+			RpmManifest: &RPMManifest{},
+			TestResults: &TestResults{},
+			Artifacts:   []Artifact{},
+		}
+	})
+
 	Context("when a project is submitted", func() {
-		BeforeEach(func() {
-			pyxisClient = NewPyxisClient("my.pyxis.host/api", "my-spiffy-api-token", "my-awesome-project-id", &http.Client{Transport: localRoundTripper{handler: mux}})
-		})
 		Context("and it is not already In Progress", func() {
 			It("should switch to In Progress", func() {
-				certResults, err := pyxisClient.SubmitResults(ctx, &certificationInput{
-					CertProject: &CertProject{CertificationStatus: "Started"},
-					CertImage: &CertImage{
-						Repositories: []Repository{
-							{
-								Registry:   "my.registry",
-								Repository: "my/repo",
-							},
-						},
-					},
-					RpmManifest: &RPMManifest{},
-					TestResults: &TestResults{},
-					Artifacts:   []Artifact{},
-				})
+				certResults, err := pyxisClient.SubmitResults(ctx, &certInput)
 				Expect(err).ToNot(HaveOccurred())
 				Expect(certResults).ToNot(BeNil())
 				Expect(certResults.CertProject).ToNot(BeNil())
 				Expect(certResults.CertImage).ToNot(BeNil())
 				Expect(certResults.TestResults).ToNot(BeNil())
 			})
 		})
-	})
-
-	Context("when a project is submitted with an empty registry", func() {
-		BeforeEach(func() {
-			pyxisClient = NewPyxisClient("my.pyxis.host/api", "my-spiffy-api-token", "my-awesome-project-id", &http.Client{Transport: localRoundTripper{handler: mux}})
+		Context("and the certImage does not have repositories", func() {
+			JustBeforeEach(func() {
+				certInput.CertImage = &CertImage{
+					Repositories: []Repository{},
+				}
+			})
+			It("should throw an error", func() {
+				certResults, err := pyxisClient.SubmitResults(ctx, &certInput)
+				Expect(err).To(HaveOccurred())
+				Expect(certResults).To(BeNil())
+			})
 		})
-		Context("and it is not already In Progress", func() {
+		Context("and a server error occurs", func() {
+			JustBeforeEach(func() {
+				pyxisClient.ApiToken = "my-error-project-api-token"
+			})
+			It("should handle the project update error", func() {
+				certResults, err := pyxisClient.SubmitResults(ctx, &certInput)
+				Expect(err).To(HaveOccurred())
+				Expect(certResults).To(BeNil())
+			})
+		})
+		Context("and the client sends a bad token", func() {
+			JustBeforeEach(func() {
+				pyxisClient.ApiToken = "my-bad-project-api-token"
+			})
+			It("should get an unauthorized", func() {
+				certResults, err := pyxisClient.SubmitResults(ctx, &certInput)
+				Expect(err).To(HaveOccurred())
+				Expect(err).To(MatchError(fmt.Errorf("%w: %s", errors.New("error calling remote API"), "could not retrieve project")))
```

Comment:
I mentioned this in Adam's PR as well, but the MatchError feels too granular in that we have the entire error definition written here, and I'm assuming, in the calling code as well.

Assuming these errors are our errors (and not errors coming from the remote API), should we be extracting these into local/package error variables? I don't think they need to go to the top level errors package, but wouldn't it be better to have the string definition of this error in one place? 

I'm not personally offended by checking that an error occurred without asserting the value of it... but I can see it either way. We're going through the effort to wrap, so if you want to test that we get the proper value, that's fine. 

Just my 2c. Feel free to disregard.

Reply:
I'm not a fan of matching the error either. I'm just going to remove.

Reply:
Why don't you all like matching the error? To me it makes sense to match the error so we know that the code is covered and hitting the proper flow we expect. We match values on `happy` path testing, ie `registry = docker.io` as an example to make sure we did something properly. I really don't understand not wanting to do the same for exceptions.

Reply:
@acornett21 for me, it's because the message we're testing for is a message that we set. I think it would be one thing if we were testing the external systems error messages (as in, make sure _this code path_ results in _that error_), but I don't think that's what's happening here.